include/ck/utility/readfirstlane.hpp

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2022, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include "ck/ck.hpp"
+#include "ck/utility/functional2.hpp"
+#include "ck/utility/math.hpp"
+
+#include <cstddef>
+#include <cstdint>
+#include <type_traits>
+
+namespace ck {
+namespace detail {
+
+template <std::size_t Size>
```

Comment:
I'm concerned with introducing "std" namespace unless there is no substitute for it. Please use "index_t"

Reply:
Shall we replace all the `std::size_t` by `index_t`? Or just remove all the qualifier `std::` in the function body?